### DISCUSSION TITLE
How to set ansi colors to font in fzf header?

### DISCUSSION THREAD

**Discussion by linrongbin16** [junegunn/fzf#3332] [Q&A] [answered]
I have try to change the font color in fzf's header.

Here's an example: https://github.com/linrongbin16/fzfx.vim/blob/fa1723e8185e3c2e3d45b6937383ef283c0d3297/autoload/fzfx/vim.vim#L88

But the ansi color code is not working, it just prints:
```
:: Press [0;31mCTRL-G[0m to fuzzy search
```

![image](https://github.com/junegunn/fzf/assets/6496887/e61ce77e-672a-41ac-a914-509b3da1d58e)

How should I fix it?

**Comment by junegunn** [comment, marked as answer]
No escape char? `\x1b`

**Reply by linrongbin16** [reply]
Sorry, I'm quite new about the ansi color....

I try to use this commit to fix: https://github.com/linrongbin16/fzfx.vim/commit/bda8db2fbadd1bc76be1bdefcdb796a7c323c336

But it looks not working:
<img width="804" alt="image" src="https://github.com/junegunn/fzf/assets/6496887/99caa39d-540f-402d-90ad-bde503d8d56d">

**Reply by junegunn** [reply]
Try using double quotes

**Reply by linrongbin16** [reply]
btw, would you please tell me, how does fzf.vim add colors in header?
I found here: https://github.com/junegunn/fzf.vim/blob/1dcdb21db618055134cd611f4f5918f6d00a5df0/autoload/fzf/vim.vim#L585

It looks that `s:magenta('CTRL-E', 'Special')` will render text to vim highlight group `Special`.
But I didn't find the implementation of this function?

**Reply by linrongbin16** [reply]
Oh, double quotes solve the issue!
This commit works: https://github.com/linrongbin16/fzfx.vim/pull/22/commits/f78938bcaffb8c99f4feb59e09fcf60fb74a2b6f

<img width="1728" alt="image" src="https://github.com/junegunn/fzf/assets/6496887/270ad0c0-46ef-4055-8a1b-b8364d0190a6">

**Reply by linrongbin16** [reply]
To be more compatible with fzf.vim, I also try to implement vim's `Special` highlight group foreground color, in this commit: https://github.com/linrongbin16/fzfx.vim/pull/23/files#diff-9c5a8207e71a7ffc01c8838723ec25fffe5584eef526e8e5851e84f6d3250c3dR74

But the result seems failed:
<img width="1107" alt="image" src="https://github.com/junegunn/fzf/assets/6496887/dde172ca-dc41-486d-989a-5e1169d35077">

The `special` and `clear` color code seems good, but it just prints wrong.
Would you please help me?

**Reply by linrongbin16** [reply]
Oh, there's a bug in string concat, fixed in this commit: https://github.com/linrongbin16/fzfx.vim/pull/23/commits/57d3e398bc19c9bb8137f2350108ce54b64c0c8b, and it worked! Great!
